src/common/transformations/src/transformations/sdpa_to_paged_attention/state_management_pattern.cpp

```diff
@@ -374,11 +391,32 @@ ov::pass::StateManagementPattern::StateManagementPattern(ParameterVector& kv_par
                                                                           max_context_len.get_node_shared_ptr()};
         pa_arguments.insert(pa_arguments.end(), additional_params.begin(), additional_params.end());
 
-        if (use_block_indices_inputs) {
+        if (use_per_layer_block_indices_inputs) {
             auto block_indices = setName(std::make_shared<v0::Parameter>(element::i32, PartialShape{-1}),
                                          "block_indices." + std::to_string(layer_index - 1));
             pa_arguments.insert(pa_arguments.begin() + 7, block_indices);
-            block_indices_inputs.push_back(block_indices);
+            block_indices_inputs_for_each_layer.push_back(block_indices);
+        }
+
+        OPENVINO_ASSERT(pa_arguments.size() == 13);
+
+
+
+        if (allow_cache_rotation) {
+            auto rotation_coefficients = setName(std::make_shared<v0::Parameter>(element::f32, PartialShape{-1}),
+                                                 "rotation_coefficients." + std::to_string(layer_index - 1));
+            auto rotated_block_indices = setName(std::make_shared<v0::Parameter>(element::i32, PartialShape{-1}),
+                                                 "rotated_block_indices." + std::to_string(layer_index - 1));
+
+            pa_arguments.insert(pa_arguments.begin() + 13, rotation_coefficients);
+            pa_arguments.insert(pa_arguments.begin() + 14, rotated_block_indices);
+
+            rotation_coefficients_inputs_for_each_layer.push_back(rotation_coefficients);
+            rotated_block_indices_inputs_for_each_layer.push_back(rotated_block_indices);
+        }
+        else {
+            pa_arguments.insert(pa_arguments.begin() + 13, v0::Constant::create(element::f32, Shape{0}, {}));
+            pa_arguments.insert(pa_arguments.begin() + 14, v0::Constant::create(element::i32, Shape{0}, {}));
```

Comment:
If you make these input really optional, this two lines are not required.

Reply:
Done